api/campaigns/class-maybe-show-campaign.php

```diff
@@ -238,15 +286,31 @@ public function should_campaign_be_shown( $client_id, $campaign, $settings, $ref
 			}
 		}
 
-		if ( ! $view_as_spec ) {
-			if ( ! empty( array_diff( $init_campaign_data, $campaign_data ) ) ) {
-				$this->save_campaign_data( $client_id, $campaign->id, $campaign_data );
-			}
-			if ( 'once' === $frequency && $campaign_data['count'] >= 1 ) {
-				$should_display = false;
-			}
-			if ( 'daily' === $frequency && $campaign_data['last_viewed'] >= strtotime( '-1 day', $now ) ) {
-				$should_display = false;
+		// Handle frequency.
+		$frequency = $campaign->f;
+		if ( ! empty( array_diff( $init_campaign_data, $campaign_data ) ) ) {
+			$this->save_campaign_data( $client_id, $campaign->id, $campaign_data );
+		}
+		if ( 'once' === $frequency && $campaign_data['count'] >= 1 ) {
+			$should_display = false;
+		}
+		if ( 'daily' === $frequency && $campaign_data['last_viewed'] >= strtotime( '-1 day', $now ) ) {
+			$should_display = false;
+		}
+
+		// Using "view as" feature.
+		if ( $view_as_spec ) {
+			$should_display = false;
+			if ( isset( $view_as_spec['segment'] ) && $view_as_spec['segment'] ) {
+				// Show prompts with a matching segment, or "everyone". Don't show any prompts that don't match the previewed segment.
+				if ( $view_as_spec['segment'] === $campaign->s || empty( $campaign->s ) ) {
+					$should_display = true;
+				}
+
+				// Show prompts if the view_as segment doesn't exist.
```

Comment:
Is that a possible UI state?

Reply:
Yes, if you preview as "Everyone" (no segment), `view_as_spec` is set but without a `segment` property.